### PR TITLE
Start replacing die() with proper error handling

### PR DESCRIPTION
**Pull request by lucaswerkmeister**
Bug: [T281804](https://phabricator.wikimedia.org/T281804)

---

Note: branch includes #6.

### CODE REVIEW

**Comment by lucaswerkmeister** [comment]
I think we’ve reached a good mid-way point now – some initial `die()`s are converted, and the rest are going to require more MediaWiki / Wikibase interaction (any `die()`s past this point require the test to at least set up a property). I’d say we can leave those for later PRs.

File: includes/MediaWiki/Api/EditEndpoint.php [hunk 1]
```diff
@@ -55,19 +57,32 @@ public function run() {
...
+			);
 		}
 		if (
 			!array_key_exists( 'urlReconcile', $inputReconcile ) ||
 			!preg_match( PropertyId::PATTERN, $inputReconcile['urlReconcile'] )
 		) {
-			die( '0.0.1 requires a single urlReconcile key mapped to a property id, such as P123' );
+			throw new LocalizedHttpException(
+				MessageValue::new( 'wikibasereconcileedit-editendpoint-invalid-reconcile-propertyid' )
+					->textParams( '0.0.1', 'urlReconcile' ),
```
Comment (toban) [inline includes/MediaWiki/Api/EditEndpoint.php]: Could we use $inputReconcile[self::VERSION_KEY] here already as we have validated it?

File: includes/MediaWiki/Api/EditEndpoint.php [hunk 2]
```diff
@@ -55,19 +57,32 @@ public function run() {
 		// TODO output an object that controls the reconciliations spec?
 		$inputReconcile = $request->reconcile();
 		if ( $inputReconcile === null ) {
-			die( 'Invalid reconcile JSON supplied' );
+			throw new LocalizedHttpException(
+				MessageValue::new( 'wikibasereconcileedit-editendpoint-invalid-reconcile-json' ),
+				400
+			);
 		}
+		$supportedReconciliationVersions = [ '0.0.1' ];
```
Comment (toban) [inline includes/MediaWiki/Api/EditEndpoint.php:66]: Not really a part of this task but i could see this moving to some other place soon.

**Review by toban — APPROVED** [review]
LGTM just some thoughts

File: includes/MediaWiki/Api/EditEndpoint.php (hunk 1 above)
Comment (lucaswerkmeister) [reply includes/MediaWiki/Api/EditEndpoint.php]: Sure, makes sense.

**Pull request merged**